### ISSUE TITLE
atom 1.5 - reset color for git modified files

### ISSUE THREAD

**Issue by albertosantini** [atom/tree-view#725]
In a git project the modified files are yellow.

If the editor loses the focus and gets it again, the files color is reset to the neutral grey.
Then, saving the same file, without any modification, you get correctly the yellow color.

The expected behavior should be the yellow color is not reset switching between the apps.
(Atom 1.5.0, tree-view 0.198.1 in safe mode)

With Atom 1.4.3 it is ok.

**Comment by hultberg** [comment]
I'm experiencing the same issue, as reported here: atom/atom#10755

**Comment by int3h** [comment]
I can confirm the bug exists in both Atom 1.5.0 and Atom 1.6.0-beta0 (tree-view 0.201.0).

**Comment by SoulGit** [comment]
Same here on 1.5.1.

**Comment by hultberg** [comment]
As of 1.5.1 and 1.6.0-beta1 I'm not having this problem, solved for me. :+1:

**Comment by albertosantini** [comment]
I confirm I have been having the same issue with 1.5.1 (on Windows 8.1).

**Comment by albertosantini** [comment]
As stated in https://github.com/atom/atom/issues/10757, there is the issue when there are multiple git projects. 

With only one git project in the tree view, it works nice.

**Comment by albertosantini** [comment]
@lee-dohm Sigh... label `needs-reproduction` means you cannot reproduce the issue?

- `atom --safe`
- no projects or files loaded
- `Add Project Folder ...` a git project
- Modify a file: the file is marked changed (yellow)
- switch to another app and switch again to Atom
- The file is correctly marked as changed
- `Add Project Folder ...` another git project (now we have two projects loaded)
- switch to another app and switch again to Atom
- Now the file of the first project is not marked as changed (grey, not yellow)

(with Atom 1.5.2 on Windows 8.1 and on Windows 10)

**Comment by nabeelomer** [comment]
What do you mean `needs-reproduction`? I lost code because of this!

**Comment by lee-dohm** [comment]
`needs-reproduction` generally means that we need repro steps added to the Issue, whether by a maintainer or by someone else.

It is also a label that our triage and QA folks search for to go back and try to reproduce so high quality repro steps can be added. I get hundreds of notifications a day across 190+ public Atom repositories. I can't deep dive investigate them all, so I add the `needs-reproduction` label so that anyone on the team can find these kinds of Issues easily.

In general, if you want things to get fixed, especially if you want it fast, it is best to supply repro steps. I [wrote a guide on my blog](http://www.lee-dohm.com/2015/01/04/writing-good-bug-reports.html) on how to write a good set of repro steps. It is simply a question of scale ... there are many many more people in the community than there are maintainers :grinning:

@albertosantini Thanks for the repro steps. I can't promise that I'll get to validating this today ... but the fact that the label is there will help bring attention to it.

**Comment by c58** [comment]
The same issue just after upgrade to 1.5.0 from previous stable version.

**Comment by joshaber** [comment]
We just released 1.5.3 which fixes this. Thanks for the reports and patience!

![](http://www.reactiongifs.com/wp-content/uploads/2013/04/clumsy.gif)

**Comment by jltrv** [comment]
@joshaber <3

**Comment by albertosantini** [comment]
Tested. It is ok.
Thanks a lot for your efforts.

**Comment by jltrv** [comment]
Works here too.

**Comment by nabeelomer** [comment]
Thank you so much!

**Comment by ChiggerChug** [comment]
This seemed closely enough related to post in this issue but I'm having a similar problem. (I'm using 1.5.4) When I commit to my repo the file color never changes in atom. It's been like this for a few days now and the issue has persisted. So the files that turn yellow stay yellow. And the files that turn green stay green.

**Comment by 50Wliu** [comment]
@ChiggerChug Can you create a new issue please?

**Comment by rodrigoaraujoti** [comment]
Same here in 1.9.0, using Ubuntu.

**Comment by htnguy02** [comment]
I just noticed the issue of modified files not being highlighted yellow in tree view as it was previously.  I'm on v1.9.1 on OSX.

**Comment by darQrrR** [comment]
We got the same issue here on different machines, running 1.9.0 on Win10.

**Comment by HMUDesign** [comment]
Seems to have come back. I was using 1.8.(something) and did not have it, but at 1.9.2, I do. (On 1.8, I did have #870 however).

**Comment by csarnataro** [comment]
The same is happening to me on OS X Yosemite with Atom 1.9.2.

Versions:
- OS: OS X Yosemite
- Atom: 1.9.2

Reproduction steps: 
- prerequisites: having a project in the tree view under git control
- open Atom. At this point the files in tree view are properly coloured regarding their git status (green for new files, orange/yellow for modified files and directory)
- cmd+tab to switch another application
- cmd+tab to come back to Atom
- all files are grey
- also, at this point, if a single file is modified, that only file change its colour in the tree view
- cmd+tab to switch another application
- cmd+tab to come back to Atom
- the modified file become grey

**Comment by hultberg** [comment]
As @50Wliu mentions: The best is to create a new issue as this might be caused by something else that this original issue was. In 1.9.0 they removed nodegit #12167 and that might be related.
